lms/djangoapps/instructor/tests/test_tools.py

```diff
@@ -284,6 +285,31 @@ def test_reset_due_date_extension_with_no_enrollment(self):
         with pytest.raises(tools.DashboardError):
             tools.set_due_date_extension(self.course, self.week3, user, extended)
 
+    @patch('edx_when.api.get_dates_for_course', wraps=get_dates_for_course)
+    def test_set_due_date_extension_cache_invalidation(self, mock_method: MagicMock):
+        """
+        Tests that the course dates are reloaded once they are overridden.
+        """
+        extended_hw = datetime.datetime(2013, 10, 25, 0, 0, tzinfo=UTC)
+        tools.set_due_date_extension(self.course, self.assignment, self.user, extended_hw)
+
+        assert mock_method.call_count == 2
+        mock_method.assert_called_with(self.course.id, user=self.user, published_version=None, use_cached=False)
```

Comment:
@Agrendalath What do you think of testing this based on the behavior instead of relying on the implementation? In other words:
1. Set up a course with dates
2. Check that `get_dates_for_course` returns the set dates (an as a side effect sets the cache)
3. Set due date extensions
4. Call `get_dates_for_course` without passing `use_cached=False`, and validate that it's returning the new dates which you set in step 3

Unless the setup would be too tedious (i.e. you have to manually initialize the in-memory cache, and patch it into the `edx_when`), I think this method of testing is preferable.

Reply:
@Cup0fCoffee, I was initially planning to use data from `edx-when`, but then saw that these tests have a custom [cache management](https://github.com/openedx/edx-platform/blob/480659b2cef84b1aabc90c7cdb844ca144e1b7d1/lms/djangoapps/instructor/tests/test_tools.py#L232-L243) that relies on the [DateLookupFieldData](https://github.com/openedx/edx-platform/blob/480659b2cef84b1aabc90c7cdb844ca144e1b7d1/lms/djangoapps/instructor/tests/test_tools.py#L195). Therefore, I decided not to test the library's behavior, but focus on making the correct calls.
I will look into adding tests that use the `edx-when` API to `lms/djangoapps/instructor/tests/test_api.py` instead. Does it make sense?

Reply:
@Agrendalath
> Does it make sense?

Yep, thanks for explanation.